awx/main/tasks.py

```diff
@@ -429,46 +429,57 @@ def execution_node_health_check(node):
 
 
 def inspect_execution_nodes(instance_list):
-    node_lookup = {}
-    for inst in instance_list:
-        if inst.node_type == 'execution':
-            node_lookup[inst.hostname] = inst
+    with advisory_lock('inspect_execution_nodes_lock', wait=True):
```

Comment:
I'm not against using an advisory lock here, it is, after all, an action global to all execution nodes. But I am very strongly in disagreement with `wait=True`, because as we scale up clusters, more and more time will be spent with a worker tied up running this task. More dangerous, it delays the further execution of the cluster_node_heartbeat, making data in the `instance_list` data structure more and more outdated.

We only _want_ this to run once per period _for the entire cluster_, so `wait=False` would have no downside.

The lock is also a little distracting. The cluster coordination problem here is _mainly_ solved by using the `last_seen` timestamp, which returns the same value throughout the receptor mesh, and updates minutely. That is how we prevent running health checks more than once per minute, per execution node.